### DISCUSSION TITLE
Metric suggestions

### DISCUSSION THREAD

**Discussion by yidilozdemir** [SciKit-Surgery/sustainable-pkg-stats#33] [Ideas]
Going over the survey from the ARC meeting, 
https://liveuclac-my.sharepoint.com/:x:/g/personal/zcbtyio_ucl_ac_uk/Ebtqz_U2NsdPlUNr7O6j06UB--xvD8cu1UeS1PxSwBgeGA?e=ayBVRc
 
I want to share a condensed version of what metrics are currently suggested both from the survey and discussions: (I will go in detail  in comments) 
- number of package dependents 
- activity metric for issues, merged PRs, commits
- lines of code/comments

Other feedback:
- definition of metrics in dropdown/click
- a simplified dashboard option with fewer metrics
- reorder the metrics so total weeks up is at the end

**Comment by yidilozdemir** [comment]
activity metric: First we need to think if we actually have packages that are active or we want to emphasize their activity. If so, we need to find formulas to represent this, here are some raised suggestions
1. issues metric: how many issues are open, average time it takes for issues to be closed
2. PR metric: number of merged PRs in last year/month, average time it takes for PR to be merged

**Comment by yidilozdemir** [comment]
number of package dependants:
I can see it being useful, especially considering the CI checks passing for most of the packages (as @thompson318) are possibly with dependencies that are not up-to-date. So in a way, having this metric could signpost to the user that as many dependencies as there are, a higher chance that they might not be up to date.
Not entirely sure how we could get this data besides, after cloning the repository, (if py) we could read a project.toml, requirements.txt or any other environment file, if it exists, and count the lines.

**Reply by thompson318** [reply]
For all our python projects we have requirements.txt so could use that. It gets a bit trickier the more generic we hope to make the dashboard though.

**Reply by yidilozdemir** [reply]
Combining this with the feedback we got for a simplified version, one solution could be to first display a more generic/simple dashboard including all the packages, then in a dropdown menu, or click, for a given package, we display in one page more metrics, including number of package dependants

**Comment by yidilozdemir** [comment]
We also mentioned number of contributors, that can be helpful as well
